portfolio/src/main/java/com/google/sps/translation/TranslationProvider.java

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.translation;
+
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+
+/** Class that allows to translate text. */
+public class TranslationProvider {
+  /**
+   * Returns a Translate object.
+   */
+  public static Translate getTranslateService() {
+    return TranslateOptions.getDefaultInstance().getService();
+  }
+
+  /**
+   * Returns the translated text to the language specified.
+   */
+  public static String translate(String text, String languageCode) {
```

Comment:
I like this interface since it does exactly one thing.

However I recommend not using static methods. Your current implementation creates a new Translate object for each translation while you could keep that across multiple translations. Right now a new one is instantiated for every comment that is translated.

Try to instead create `Translate` once and store it in the instance on a field. Then the translate method can reuse that.

Reply:
Ok, I can use the singleton pattern.

Reply:
I wasn't trying to make you use the singleton pattern. But rather create an instance knowingly in the Servlet and then use that. The Singleton pattern comes with a lot of drawbacks if implemented manually. This is a long article but captures the important parts: https://www.vojtechruzicka.com/singleton-pattern-pitfalls/#:~:text=One%20of%20the%20main%20disadvantages,test%20the%20Singleton%20as%20well.
I.e. testing is harder, dependencies become hidden, concurrency is harder

Reply:
I wasn't aware of these drawbacks. Thanks!